### ISSUE TITLE
Problem getting the migrations inside a OTP release

### ISSUE THREAD

**Issue by reimo** [ateliware/triplex#41]
Hi, 
Tried moving an app from heroku to a vps 
Deployment using gatling and everything runs well except Triple  Triplex.create("tenant") returns {:ok, []) on production.

The app is still running heroku and everything is great.

Platform
Ubuntu 16.04
Erlang/OTP 19 [erts-8.3.5] [source] [64-bit] [async-threads:10] [hipe] [kernel-poll:false]
Elixir 1.4.5
Deployment with gatling

**Comment by kelvinst** [comment]
> Triplex.create("tenant") returns {:ok, []) on production.

Sorry, but what was the return you were expecting for it?

**Comment by kelvinst** [comment]
Just to mention: on `Triplex.create/2`, we actually just return what `Ecto.Adapters.SQL.query/3` returns, [as you can see on the code](https://github.com/ateliware/triplex/blob/master/lib/triplex.ex#L206-L215). 

That's probably what's changing the function result after changing the server, which probably also changed postgres version.

I'll think a little bit more about it, but probably we must change this function return to something that does not change on different postgres versions!

Thanks for the feedback BTW.

**Comment by reimo** [comment]
Ok, Let me know if i can help fasten the process 

N/B I have posgresql 9.6 locally and online

**Comment by kelvinst** [comment]
Working on the fix now, we will always return `{:ok, tenant}` where tenant is the given tenant param

**Comment by kelvinst** [comment]
@reimo the fix is merged on master, but we will release it on the next minor version since it breaks compatibility with the previous behaviour.

You can either use it on the edge version by referencing this repo on you mix.exs with `{:triplex, github: "ateliware/triplex", branch: "master"}`, or wait until the next release, which will be done probably until the end of this week.

**Comment by reimo** [comment]
@kelvinst guess i was not able to explain the situation properly.

So here i go.
I seed my newly created tenant immediately after creating them.

Here is the code 

```
tenant = "tenant23"
 case Triplex.create(tenant) do
           {:ok, _} -> 
                      #this case is true but in reality the tenant was not created which make the next line fail.
                      seed = RexCore.House.Seeder.seed(tenant)
                      {:ok, school, seed}
           {:error,  _} = error ->
                   {:error, error}
   end
```
 
```
@doc """
  Creates the given `tenant` on the given `repo`.
  Besides creating the database itself, this function also loads their
  structure executing all migrations from inside
  `priv/YOUR_REPO/tenant_migrations` folder.
  """
  def create(tenant, repo \\ config().repo) do
    create_schema(tenant, repo, &(migrate(&1, &2)))
  end
```
 so the issue is that the database / schema was not created on this new server but Triplex.create(tenant) still returned {:ok, []}


hope this explains it

**Comment by kelvinst** [comment]
I see! Reopening the issue! I'll investigate it ASAP...

**Comment by kelvinst** [comment]
@reimo, what is the error you get when seeding? 

I'm trying to repeat the problem but I really don't know which problem is it.

PS.: Looks like `{:ok, []}` is the return of `migrate/2`, which means that no migration were run inside the tenant schema. Please, make sure your tenant migrations are inside the `priv/repo/tenant_migrations` folder, and not inside `priv/repo/migrations`.

**Comment by reimo** [comment]
i have migration files in  /priv/repo/migrations
On the new server i used gatling and Distillery to achieve heroku like push to deployment

Thinks i dont know, 1 if gatling or distillery changes the directory structure
Please try using gatling to confirm if u have the same effect.

**Comment by kelvinst** [comment]
So, I'm not sure I understand, what are you inserting in this seed? What kind of error are you getting for seeding?

Because if all your migrations are in `priv/repo/migrations` there is no table to be created inside the tenant database. The database is being created but no migrations ran, that's why the result is `{:ok, []}`, the second element in this tuple is an array with the timestamps of the migrations that ran. And if you don't have any migration on `priv/repo/tenant_migrations`, that's just right.

I suspect the error on seeding is due to this situation, you probably have a migration on `priv/repo/migrations` which needs to be in `priv/repo/tenant_migrations`.

Can you please share with us the error you are getting and maybe the tree structure of your `priv/repo` folder?

**Comment by reimo** [comment]
sorry i am responding to this late 
<img width="201" alt="screen shot 2017-11-07 at 7 00 38 am" src="https://user-images.githubusercontent.com/6754050/32479322-8b7a59f6-c389-11e7-8e26-bbf7b94a5671.png">

the point is that everything is in their rightful place, the application works perfectly ok on heroku but returns {:ok, []} on vps deployed with gatling.

i did inspect my build priv/repo/tenant_migrations and the migration is intact.
so having {:ok, []} when everything looked was the reason for the issue.

**Comment by kelvinst** [comment]
I see now! Thanks for the responses and sorry for bothering that much. So the problem is probably related to distillery not copying the migrations folders when creating the release. 

Can you please check if the `tenant_migrations` folder is inside your release?

I'ld do it myself, but I am without much free time these days, but if you're not in a hurry I can do it next week probably.

**Comment by reimo** [comment]
@kelvinst  Yes i have the migration, the picture above was from my build.

**Comment by kelvinst** [comment]
So the problem is for sure with triplex way to get the migration files inside distillery releases! :) 

Thanks for all the infos, I'll create a test project for it and fix the problem!

**Comment by kelvinst** [comment]
@reimo Can you share your ecto version?

**Comment by reimo** [comment]
Ecto version 2.2.6

**Comment by kelvinst** [comment]
@reimo just remembering to check if #49 fixed this! :)

**Comment by kelvinst** [comment]
@reimo, I'm opening a `1.2.0-rc` release, it will be a little better for you to test ;)

**Comment by kelvinst** [comment]
@reimo I've released the `1.2.0` official review with the fixes. Closing the issue.
